File: app/src/main/java/fr/frogdevelopment/nihongo/dico/ui/BottomNavigationDrawerFragment.java
package fr.frogdevelopment.nihongo.dico.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.google.android.material.navigation.NavigationView;

import fr.frogdevelopment.nihongo.dico.R;
import fr.frogdevelopment.nihongo.dico.databinding.BottomsheetFragmentBinding;

public class BottomNavigationDrawerFragment extends BottomSheetDialogFragment implements NavigationView.OnNavigationItemSelectedListener {

    private final OnNavigateToListener onNavigateToListener;
    private BottomsheetFragmentBinding mBinding;

    public BottomNavigationDrawerFragment(OnNavigateToListener onNavigateToListener) {
        super();
        this.onNavigateToListener = onNavigateToListener;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NORMAL, R.style.NavigationBottomSheet);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mBinding = BottomsheetFragmentBinding.inflate(getLayoutInflater());
        mBinding.navigationView.setNavigationItemSelectedListener(this);

        return mBinding.getRoot();
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        if (onNavigateToListener != null) {
            onNavigateToListener.navigateTo(item.getItemId());
        }

        dismiss();
        return true;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mBinding = null;
    }

    interface OnNavigateToListener {

        void navigateTo(int itemId);
    }

}

File: app/src/main/java/fr/frogdevelopment/nihongo/dico/ui/BottomNavigationDrawerFragment.kt
package fr.frogdevelopment.nihongo.dico.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.google.android.material.navigation.NavigationView
import fr.frogdevelopment.nihongo.dico.R
import fr.frogdevelopment.nihongo.dico.databinding.BottomsheetFragmentBinding

class BottomNavigationDrawerFragment(private val onNavigateToListener: OnNavigateToListener?) : BottomSheetDialogFragment(), NavigationView.OnNavigationItemSelectedListener {

    private var _binding: BottomsheetFragmentBinding? = null
    // This property is only valid between onCreateView and onDestroyView.
    private val binding get() = _binding!!

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(DialogFragment.STYLE_NORMAL, R.style.NavigationBottomSheet)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        _binding = BottomsheetFragmentBinding.inflate(layoutInflater)
        binding.navigationView.setNavigationItemSelectedListener(this)
        return binding.root
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        onNavigateToListener?.navigateTo(item.itemId)
        dismiss()
        return true
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    interface OnNavigateToListener {
        fun navigateTo(itemId: Int)
    }

}